cpp/test/Ice/optional/Test.ice

```diff
@@ -278,9 +278,9 @@ interface Initial
 
     void opClassAndUnknownOptional(A p);
 
-    void sendOptionalClass(bool req, optional(1) OneOptional o);
+    void sendOptionalStruct(bool req, optional(1) FixedStruct ofs);
```

Comment:
We already have extensive testing for struct above, with opSmallStruct, opFixedStryct, opVarStruct and even tests for struct list.

What are these newly renamed operations sendOptionalStruct and returnOptionalStruct adding in terms of test coverage?

It's ok to delete redundant tests.

Reply:
After reading the comment in the .cpp file, I think you missed the intent of this test.

It's not about passing an optional parameter, but about passing a class parameter with optional fields. The additional layer of making the parameter optional is superfluous.

So I would keep the names and simplify the signature:
```slice
void sendOneOptionalClass(OneOptional o);
...

Reply:
If you want to remove it, that's fine with me, I don't really care.
But, I get the opposite idea from reading the original comment:
```
// Use the 1.0 encoding with operations whose only class parameters are optional.
```
It doesn't say alot. But it sounds like the point is "... parameters are optional"
and it doesn't mention anything about the fields that class holds.

I tracked down the original issue: https://zeroc.atlassian.net/browse/ICE-4930
I don't really understand what's going on though. It's from 3.4.2 apparently.

Reply:
Looking at the initial commit that fixes this https://github.com/zeroc-ice/ice/commit/493caea19fd83663eb03b4eeaf7714f99ac07b97

I think this test was about the class encoding, more concretely about a bogus call to `writePendingObjects` 

https://github.com/zeroc-ice/ice/commit/493caea19fd83663eb03b4eeaf7714f99ac07b97#diff-a6f653e19cbe0e6ef7bfee702bbb838ccd86ce17d0454083c81e670e4a8020cdL2173-L2176

If we are only sending optional classes seems we don't have to call the method, but be were calling it.

The PR added this additional `includeOptional` parameter to the  `Slice::Operation::sendsClasses` operation.

I think we can remove it, as it was class encoding related and the new version doesn't use classes.